packages/modal/src/Modal/Modal.tsx

```diff
@@ -45,18 +47,30 @@ const Root = ({
   return <RootComponent open={open} onOpenChange={onOpenChange} {...props} />;
 };
 
+type RootComponentProps = RootProps & {
+  open: boolean;
+  onOpenChange: (value: boolean) => void;
+};
+
 const RootComponent = ({
   open,
   onOpenChange,
   a11yTitle,
   a11yDescription,
   children,
   layer: layerProp,
+  // Focus scope props
+  trapped,
+  onMountAutoFocus,
+  onUnmountAutoFocus,
```

Comment:
Allowed for FocusScope props to be passed so we can have better control of the focused element inside the modal.